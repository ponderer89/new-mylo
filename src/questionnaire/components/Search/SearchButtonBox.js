import React from 'react'
import styled from 'styled-components/macro'
import SingleInput from '../SingleInput/SingleInput'

const StyledSingleInput = styled(SingleInput)`
  bottom: 0;
`

const SearchButtonBox = ({ value, onChange }) => {
  return (
    <StyledSingleInput
      value={value}
      onChange={onChange}
      placeholder="Start typing company name"
    />
  )
}

export default SearchButtonBox
