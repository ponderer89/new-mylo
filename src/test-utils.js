import React from 'react'
import { Provider } from 'react-redux'
import { ThemeProvider } from 'styled-components/macro'
import { createStore } from './store'
import theme from './theme'
import { render } from 'react-testing-library'

export const r = C => {
  const store = createStore()

  return {
    ...render(
      <Provider store={store}>
        <ThemeProvider theme={theme}>{C}</ThemeProvider>
      </Provider>
    ),
    store
  }
}
