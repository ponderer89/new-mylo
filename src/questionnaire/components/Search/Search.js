import React, { Fragment, useState, useCallback } from 'react'
import RadioField from '../RadioField/RadioField'
import SearchButtonBox from './SearchButtonBox'

const Search = ({ name, options, value, onSelect }) => {
  const [searchInputText, setSearchInputText] = useState('')
  const onChange = useCallback(e => {
    setSearchInputText(e.target.value)
  }, [])

  const filteredOptions = options.filter(
    o =>
      o.label.startsWith(searchInputText) || o.value.startsWith(searchInputText)
  )

  return (
    <Fragment>
      <RadioField
        name={name}
        options={filteredOptions}
        value={value}
        onChange={onSelect}
      />
      <SearchButtonBox value={searchInputText} onChange={onChange} />
    </Fragment>
  )
}

export default Search
