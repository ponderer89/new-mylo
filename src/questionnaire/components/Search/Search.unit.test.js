import React from 'react'
import { r } from 'test-utils'
import Search from './Search'
import { fireEvent } from 'react-testing-library'

const OPTIONS = [
  {
    label: 'label',
    value: 'value'
  }
]

describe('Search', () => {
  it('should render', () => {
    const { container } = r(
      <Search name="name" value="value" options={OPTIONS} onSelect={jest.fn} />
    )

    expect(container).toMatchSnapshot()
  })

  it('should filter the options when using the search input', () => {
    const _OPTIONS = [{ label: 'A', value: 'A' }, { label: 'B', value: 'B' }]

    const { container, queryByText } = r(
      <Search name="name" value="value" options={_OPTIONS} onSelect={jest.fn} />
    )

    fireEvent.change(container.querySelectorAll('input')[2], {
      target: { value: 'B' }
    })

    expect(queryByText('A')).not.toBeInTheDocument()
  })
})
