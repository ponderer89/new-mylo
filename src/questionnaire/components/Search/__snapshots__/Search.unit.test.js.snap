// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render 1`] = `
<div>
  <ul
    class="RadioField__StyledRadioField-dbgo0q-0 GXYln"
  >
    <li
      class="RadioItem__StyledRadioItem-sc-1e6z0nt-0 kGzXTL"
    >
      <input
        checked=""
        name="name"
        type="radio"
        value="value"
      />
      <label>
        label
      </label>
    </li>
  </ul>
  <div
    class="SingleInput__StyledSingleInput-sc-1titmac-0 lmDrXh"
  >
    <div
      class="NextButton__StyledButtonBox-ar7dhl-0 SearchButtonBox__StyledSingleInput-sc-1k3go0o-0 ctzWkS SingleInput__StyledInputBox-sc-1titmac-1 gtspmL"
    >
      <input
        class="SingleInput__StyledInput-sc-1titmac-2 byFhVL"
        placeholder="Start typing company name"
        value=""
      />
    </div>
  </div>
</div>
`;
